### DISCUSSION TITLE
How to save JPG file to array for reuse

### DISCUSSION THREAD

**Discussion by LarsMachiels** [Bodmer/TFT_eSPI#989] [Q&A - General] [answered]
I am trying to figure out how I can save a JPG from ESP32 SPIFFS into an array so I can reuse it without reloading. 
Any help and/or sample code is highly appreciated!

**Comment by Bodmer** [comment, marked as answer]
The header comments [here](https://github.com/Bodmer/TJpg_Decoder/blob/master/examples/Flash_Jpg/panda.h) tell you how to convert a file into an array and then it can be kept in FLASH instead of SPIFFS.

Typically Jpeg files are highly compressed and most processors do not have the space to store large decompressed image arrays.

Another approach is to avoid an array and render the jpeg into a sprite, then save the meory block.
